internal/services/containerapps/container_app_environment_resource.go

```diff
@@ -0,0 +1,370 @@
+			if containerAppEnvironment.InfrastructureSubnetId != "" {
+				managedEnvironment.Properties.VnetConfiguration.InfrastructureSubnetId = utils.String(containerAppEnvironment.InfrastructureSubnetId)
+				managedEnvironment.Properties.VnetConfiguration.Internal = utils.Bool(containerAppEnvironment.InternalLoadBalancerEnabled)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, managedEnvironment); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ContainerAppEnvironmentResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ContainerApps.ManagedEnvironmentClient
+			id, err := managedenvironments.ParseManagedEnvironmentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(existing.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			var state ContainerAppEnvironmentModel
+
+			if model := existing.Model; model != nil {
+				state.Name = id.EnvironmentName
+				state.ResourceGroup = id.ResourceGroupName
+				state.Location = location.Normalize(model.Location)
+				state.Tags = tags.Flatten(model.Tags)
+
+				if props := model.Properties; props != nil {
+					if vnet := props.VnetConfiguration; vnet != nil {
+						state.InfrastructureSubnetId = utils.NormalizeNilableString(vnet.InfrastructureSubnetId)
+						state.InternalLoadBalancerEnabled = utils.NormaliseNilableBool(vnet.Internal)
+						state.DockerBridgeCidr = utils.NormalizeNilableString(vnet.DockerBridgeCidr)
+						state.PlatformReservedCidr = utils.NormalizeNilableString(vnet.PlatformReservedCidr)
+						state.PlatformReservedDnsIP = utils.NormalizeNilableString(vnet.PlatformReservedDnsIP)
+					}
+
+					state.StaticIP = utils.NormalizeNilableString(props.StaticIP)
+					state.DefaultDomain = utils.NormalizeNilableString(props.DefaultDomain)
+				}
+
+				if sysData := model.SystemData; sysData != nil {
+					state.CreatedAt = sysData.CreatedAt
+					state.CreatedBy = sysData.CreatedBy
+					state.CreatedByType = sysData.CreatedByType
+					state.LastModifiedAt = sysData.LastModifiedAt
+					state.LastModifiedBy = sysData.LastModifiedBy
+					state.LastModifiedByType = sysData.LastModifiedbyType
+				}
+			}
+
+			// Reading in log_analytics_workspace_id is not possible, so reading from config. Import will need to ignore_changes unfortunately
+			if v := metadata.ResourceData.Get("log_analytics_workspace_id").(string); v != "" {
+				state.LogAnalyticsWorkspaceId = v
+			}
+
+			if err := metadata.Encode(&state); err != nil {
+				return fmt.Errorf("encoding: %+v", err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ContainerAppEnvironmentResource) Delete() sdk.ResourceFunc {
```

Comment:
Is it possible to keep track of the separate resource group that is created and not return before it is deleted? Otherwise a dependency to the virtual network (subnet) will still be used and the destroy of it wont pass until the resource group is removed.

Reply:
Hi @simongottschlag -  It _sounds_ like you may have an ordering/dependency issue in your implementation via the azapi provider?  For completeness I've added a test [here](https://github.com/hashicorp/terraform-provider-azurerm/pull/18008/files#diff-3dc98b526f8f6548c75e66c74ec0ec33e222cca8cc4f64ac40d139f34e37fc17R86) which builds everything, including Vnet & subnet, in one run which has passed every time so far.